# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/QuicSocketControl.cpp
# Commit: 496ec0c5a60f
# Full Hash: 496ec0c5a60fdb86b233d036dd4747c477fc33ca
# Author: Dragana Damjanovic <dd.mozilla@gmail.com>
# Date: 2019-10-31 09:53:09
# Regressor Bug: 1581637
# File Overlap Count: 1
# Description:
#   Bug 1581637 - Part 8 - Add Http3Session/Http3Stream. r=mayhemer
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D46652
# ==============================================================================

diff -r 63b7f1ff1714 -r 496ec0c5a60f netwerk/protocol/http/QuicSocketControl.cpp
--- a/netwerk/protocol/http/QuicSocketControl.cpp	Wed Oct 30 20:54:21 2019 +0000
+++ b/netwerk/protocol/http/QuicSocketControl.cpp	Wed Oct 30 21:09:09 2019 +0000
@@ -43,20 +43,17 @@
 }
 
 void QuicSocketControl::CallAuthenticated() {
-  //  // Will be added when Http3 lands
-  /*  if (mHttp3Session) {
+  if (mHttp3Session) {
     RefPtr<Http3Session> http3Session = do_QueryReferent(mHttp3Session);
     http3Session->Authenticated(GetErrorCode());
   }
-  mHttp3Session = nullptr;*/
+  mHttp3Session = nullptr;
 }
 
-// Will be added when Http3 lands
-// void QuicSocketControl::SetAuthenticationCallback(Http3Session
-// *aHttp3Session) {
-//  mHttp3Session = do_GetWeakReference(
-//      static_cast<nsISupportsWeakReference*>(aHttp3Session));
-//}
+void QuicSocketControl::SetAuthenticationCallback(Http3Session* aHttp3Session) {
+  mHttp3Session = do_GetWeakReference(
+      static_cast<nsISupportsWeakReference*>(aHttp3Session));
+}
 
 void QuicSocketControl::HandshakeCompleted() {
   psm::RememberCertErrorsTable::GetInstance().LookupCertErrorBits(this);